core/src/main/scala/org/bitcoins/core/wallet/utxo/AddressTag.scala

```diff
@@ -0,0 +1,14 @@
+package org.bitcoins.core.wallet.utxo
+
+case class AddressTagType(name: String)
+
+case class AddressTag(tag: String, tagTypeOpt: Option[AddressTagType])
```

Comment:
Why is it `tagTypeOpt` rather than just `tagType: AddressTagType`

Reply:
Because we don't always have access to the `tagType`, for instance the db mapper.

Reply:
In that case I think it would be nice to have an intermediate state without the `AddressTagType` which has a `toAddressTag` method which takes in an `AddressTagType`